tasks/set_vars.yml

```diff
@@ -1,21 +1,27 @@
 ---
-- name: Ensure ansible_facts used by role
-  setup:
-    gather_subset: min
-  when: not ansible_facts.keys() | list |
-    intersect(__template_required_facts) == __template_required_facts
+- name: "Gather the packages facts"
+  ansible.builtin.package_facts:
+    manager: rpm
 
-- name: Set platform/version specific variables
-  include_vars: "{{ __vars_file }}"
-  loop:
-    - "{{ ansible_facts['os_family'] }}.yml"
-    - "{{ ansible_facts['distribution'] }}.yml"
-    - >-
-      {{ ansible_facts['distribution'] ~ '_' ~
-      ansible_facts['distribution_major_version'] }}.yml
-    - >-
-      {{ ansible_facts['distribution'] ~ '_' ~
-      ansible_facts['distribution_version'] }}.yml
-  vars:
-    __vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __vars_file is file
+- name: "Check postgresql.conf"
+  stat:
+    path: /var/lib/pgsql/data/postgresql.conf
+  register: postgresql_conf
+
+- name: "Check TLS crt file"
+  stat:
+    path: "/etc/pki/tls/certs/{{ cert_name }}.crt"
+  register: crt_file
```

Comment:
use `__postgresql_crt_file` for internal vars
Also, see https://linux-system-roles.github.io/documentation/tls_crypto_parameter_and_key_names.html for the naming conventions around crypto naming. This should probably be `__postgresql_cert`